### ISSUE TITLE
Status

### ISSUE THREAD

**Issue by KM1964** [CurlyTaleGamesLLC/PyKiln#1]
Hello,  I am wondering about this projects status.

**Comment by michthom** [comment]
Same here - I've hit problems installing / running the current version of the code, and I'm hoping Ben can help. I'd be very happy to act as a tester and perhaps contributor as I'm very impressed with the videos he posted on YouTube, like the approach to the safety aspects as well as a polished user interface.

**Comment by KM1964** [comment]
Yes, I too am very impressed. I am on vacation at the moment but I am planing to try an install when I get back.  I was wondering if Ben has board ordered and if so what the time line is.  If it will be a while I may try to bread board it.

**Comment by michthom** [comment]
Have you watched the [YouTube videos on his channel](https://www.youtube.com/channel/UCMisTGQ-q9bf2-pH9ZL5CeQ/videos)? These imply that he's in the midst of a fairly significant redesign, so might be worth prototyping your implementation for now? I'm certainly heading that way.
As I've been unable to get the PyKiln version running, I've looked at the [older PiKiln repository](https://github.com/CurlyTaleGamesLLC/PiKiln) which runs on a Raspberry Pi, and am looking to see if I can overcome some of the limitations.
Ben was unhappy that the Pi doesn't have onboard ADC to read the current sensor, but as it only really needs to know if it's on or off, I tried modelling a simple circuit to take the AC current transformer output and turn that into a (near) DC GPIO input voltage, see below.
<img width="1662" alt="AC Detector" src="https://user-images.githubusercontent.com/546179/120942193-33afb980-c71f-11eb-936a-09c9e3e0b159.png">

**Comment by BenRinthLabs** [comment]
Hi! Glad you like the project, but it's not at a functional state at the moment. If you're looking to set up a kiln right now there are a couple other open source options out there, like jbruce's Raspberry Pi Kiln controller: https://github.com/jbruce12000/kiln-controller

As far as status on PyKiln, I've made another big shift in the design recently. I'm switching the PyKiln breakout board to use i2c temperature sensors, as well as an i2c GPIO expander to control the relays. 

I also have been playing around with logging to a microSD card instead of the internal flash storage on the ESP32. Logs will cause a lot of writes to the storage, and would eventually destroy the flash storage on the ESP32. Using a microSD card will allow for tons of space for logs and firing schedules. I might move the wifi configuration into a text file on the microSD card too. If you have some experience with python or Vue I'd love to collaborate :)

**Comment by steeleh** [comment]
I am attempting to blag this code enough to run a prototype. There was a problem writing the ip.txt file at set up so I added it manually and that got me up to the end of the setup with <settings> web page running and able to alter setting but the code hangs at  
* Running on http://192.168.0.63:80/
I (9626) modsocket: Initializing
INFO:picoweb:58.000 <HTTPRequest object at 3ffe6ba0> <Stream object at 3ffe6910> "GET /"

I am not Python literate: but my son is a senior software engineer and lives in Python, C++, etc. If there is a particular block that is causing a log jam I may be able to help.....

**Comment by MitchBradley** [comment]
Is this project still active?  If so I might be able to help.  I am a retired EE/programmer with experience in C, C++, Python, and JavaScript, among others.  I am a primary developer on the FluidNC project, which uses ESP32 for CNC machine control, so I know the ins and outs of ESP32 and its various frameworks.  I want to convert a friend's kiln to digital control.

**Comment by BenRinthLabs** [comment]
It's still alive, I just haven't had time to work on it. I had some very serious health issues in my family this year and PyKiln was shelved for a bit. If you need an IoT kiln controller right now, I'd recommend Jbruce's kiln controller: [https://github.com/jbruce12000/kiln-controller](https://github.com/jbruce12000/kiln-controller)

I started to make a pivot to supporting a touch screen earlier this year, so there are some big changes to the code base that I need to finish up, and commit.

Once I get that committed I'll try to clean up this repo and organize things so other people can help out

Thanks for the interest in the project!

**Comment by MitchBradley** [comment]
I suppose I will have to resurrect one of the Pi's that is gathering dust in a box, much as I dislike administering Linux.  Spent too much of my life doing that :-)
